### PR TITLE
Added simple cache for survolBackgroundRequest

### PR DESCRIPTION
**Pull request by Brett78**
Added a simple page based cache for the survolBackgroundRequest. Addresses #44 

Not sure if you wanted a more consistent cache across pages, but it seems like a page cache would be sufficient.

### CODE REVIEW

**Review by mdolr — CHANGES_REQUESTED** [review]
Hum thank you for your pull request, I forgot to mention the cache should be implemented in `js/background/auxiliary.js`. By adding it there it makes more sense as it will be shared with all pages.

Just move the cache object to auxiliary.js and fetch only if the url is not already cached. You should be able to pretty much copy paste what you've done

**Comment by Brett78** [comment]
No worries, do you have any ideas on how you would want to invalidate this cache? Or are you not concerned with this for the moment?

**Comment by mdolr** [comment]
Not really, I think a setInterval every 10 minutes clearing it entirely should do the job for now, we don't need anything fancy I'd say.

**Comment by Brett78** [comment]
Just committed some changes, let me know if there's anything else! :)

**Review by mdolr — APPROVED** [review]

**Comment by mdolr** [comment]
It should be good thanks

File: js/background/auxiliary.js [hunk 1]
```diff
@@ -12,26 +12,32 @@ setInterval(() => {
...
-            })
-            .catch((error) => {
-                res.data = error;
-                res.status = 'error';
-                sendResponse(res);
-                console.error('SURVOL - Fetching error', error);
-            });
+
+        // if the request is cached
+        if (REQUEST_CACHE[url]) {
```
Comment (Brett78) [inline js/background/auxiliary.js:19]: Does this need to be req.data.url?

Comment (mdolr) [reply js/background/auxiliary.js:19]: It does thank you I tried to rush things, should have tested :/

**Pull request merged**